### PR TITLE
feat: fixes after testing with noisier data

### PR DESCRIPTION
**Pull request by msramalho**

### CODE REVIEW

**Review by fspoettel — APPROVED** [review]

File: geocluster/clustering.py [hunk 1]
```diff
@@ -32,6 +32,14 @@ def cluster_locations(df, algorithm, radius_km, min_cluster_size):
     within `radius_km` of each other.
     Outputs a dict grouping locations by `cluster_id`.
     """
+    valid_index = df.lat.str.contains("^\d+.?\d*$", regex=True) & df.lon.str.contains(
```
[the remark this answers is not in the document]
Comment (fspoettel) [reply geocluster/clustering.py]: I think we are already dropping _some_ rows with running `dropna()` on the input set. we might want to log these (or remove the `dropna if we validate manually anyway).

[the remark this answers is not in the document]
Comment (msramalho) [reply geocluster/clustering.py]: done, please check https://github.com/fspoettel/geocluster/pull/2/commits/ab3e1fd3af1d84ebd9ab2a513c6f0eb002f33328  i could not find the in_range so feel free to refactor that, but tested in easy and bad data, and works

File: geocluster/__main__.py [hunk 2]
```diff
@@ -41,6 +42,7 @@
 @click.argument("filename", type=click.Path(exists=True))
 def main(distance, size, output, filename, algorithm):
     df = io.read_csv_file(filename)
+    df = df.replace({np.nan: None})
```
Comment (fspoettel) [inline geocluster/__main__.py:45]: Should we move this to `io.read_csv_file`?

**Review by fspoettel — APPROVED** [review]
Nice! :raised_hands: Minor questions, feel free to merge anytime.

**Pull request merged**